### PR TITLE
[bug] Disable pan drift for pitched maps

### PR DESCRIPTION
**Pull request by neelmistry94**
## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog>Pan drift for pitched maps will be disabled. A solution for smooth drifting is being worked on.</changelog>`.

### Summary of changes
This PR will temporarily disable drift for pitched maps when a user is panning. The current experience when panning a pitched maps results in the map jumping to undesired location. We are currently working on a final solution for smooth drifting.

### CODE REVIEW

File: Mapbox/MapboxMapsGestures/GestureManager.swift [hunk 1]
```diff
@@ -283,7 +283,10 @@ extension GestureManager: GestureHandlerDelegate {
 
     // Pan has ended on the MapView with a residual `offset`
     internal func panEnded(with offset: CGPoint) {
-        self.cameraManager.moveCamera(by: offset, animated: true)
+        if let pitch = self.cameraManager.mapView?.pitch,
+           pitch == 0.0 {
```
Comment (macdrevx) [inline Mapbox/MapboxMapsGestures/GestureManager.swift:287]:
```suggestion
        if cameraManager.mapView?.pitch == 0.0 {
```

File: Mapbox/MapboxMapsGestures/GestureManager.swift [hunk 2]
```diff
@@ -283,7 +283,10 @@ extension GestureManager: GestureHandlerDelegate {
...
+            self.cameraManager.moveCamera(by: offset, animated: true)
```
Comment (macdrevx) [inline Mapbox/MapboxMapsGestures/GestureManager.swift:288]:
```suggestion
            cameraManager.moveCamera(by: offset, animated: true)
```

**Review by macdrevx — APPROVED** [review]

**Review by nishant-karajgikar — APPROVED** [review]

File: Mapbox/MapboxMapsGestures/GestureManager.swift (hunk 2 above)
Comment (neelmistry94) [reply Mapbox/MapboxMapsGestures/GestureManager.swift:288]: we are using `self.` in most places so would like to keep that for consistency

File: Mapbox/MapboxMapsGestures/GestureManager.swift (hunk 1 above)
Comment (neelmistry94) [reply Mapbox/MapboxMapsGestures/GestureManager.swift:287]: I thought about this, but what if mapView is nil for some reason.

File: Mapbox/MapboxMapsGestures/GestureManager.swift (hunk 2 above)
Comment (macdrevx) [reply Mapbox/MapboxMapsGestures/GestureManager.swift:288]: Sounds good.

File: Mapbox/MapboxMapsGestures/GestureManager.swift (hunk 1 above)
Comment (macdrevx) [reply Mapbox/MapboxMapsGestures/GestureManager.swift:287]: If mapView is nil, the expression will evaluate to false.

Comment (macdrevx) [reply Mapbox/MapboxMapsGestures/GestureManager.swift:287]: Which seems fine. You'll just be avoiding the assertionFailure inside of moveCamera

Comment (macdrevx) [reply Mapbox/MapboxMapsGestures/GestureManager.swift:287]: This has the same behavior as what you wrote initially, it's just more concise.

**Pull request merged**